core-backend/src/funcs.rs

```diff
@@ -1297,21 +1298,36 @@ where
         })
     }
 
-    pub fn out_of_gas(_gas: Gas) -> impl Syscall<Ext> {
-        RawSyscall::new(|ctx: &mut CallerWrap<Ext>| {
-            let ext = ctx.ext_mut();
-            let current_counter = ext.current_counter_type();
-            log::trace!(target: "syscalls", "[out_of_gas] Current counter in global represents {current_counter:?}");
+    fn out_of_gas(ctx: &mut CallerWrap<Ext>) -> UndefinedTerminationReason {
+        let ext = ctx.ext_mut();
+        let current_counter = ext.current_counter_type();
+        log::trace!(target: "syscalls", "system_break(OutOfGas): Current counter in global represents {current_counter:?}");
 
-            if current_counter == CounterType::GasAllowance {
-                // We manually decrease it to 0 because global won't be affected
-                // since it didn't pass comparison to argument of `gas_charge()`
-                ext.decrease_current_counter_to(0);
-            }
+        if current_counter == CounterType::GasAllowance {
+            // We manually decrease it to 0 because global won't be affected
+            // since it didn't pass comparison to argument of `gas_charge()`
+            ext.decrease_current_counter_to(0);
+        }
+
+        ActorTerminationReason::from(current_counter).into()
+    }
 
-            let termination_reason: ActorTerminationReason = current_counter.into();
+    fn stack_limit_exceeded() -> UndefinedTerminationReason {
+        log::trace!(target: "syscalls", "system_break(StackLimitExceeded)");
+        TrapExplanation::StackLimitExceeded.into()
+    }
+
+    pub fn system_break(_gas: Gas, code: u64) -> impl Syscall<Ext> {
```

Comment:
btw why not u32? for codes-consistency with error codes

Reply:
Btw `i64.const` and `i32.const` has same cost. If we can have the ability to extend syscall for free in future, then why not leave `i64`?

Reply:
Just wondering atm. Anyway this call and its arg are inserted on instrumentation